### ISSUE TITLE
coredns errors, timeouts

### ISSUE THREAD

**Issue by roy-work** [Azure/AKS#1752]
**What happened**:
We saw a few thousand errors from `coredns`:
```
[ERROR] plugin/errors: 2 . NS: dns: overflow unpacking uint32
[ERROR] plugin/errors: 2 . NS: read udp 10.12.0.69:33077->168.63.129.16:53: i/o timeout
```

"overflow unpacking uint32" seems to have nothing to do with overflow; AFAICT, it is a failed bounds check during the decoding of the DNS packet; it seems that coredns chooses some rather conservative sizes for its buffer for incoming UDP packets (512 B at one point; I'm not sure what the value used by the current version of coredns in AKS is, though) resulting in the packet getting truncated.

Still, I don't think we're resolving anything except for A records, so it seems unlikely that a response should even exceed that bound.

The timeouts also indicate an outage on Azure's side.

**Environment**:
- Kubernetes version (use `kubectl version`):
- Size of cluster (how many worker nodes are in the cluster?)
- General description of workloads in the cluster: A CockroachDB cluster; I don't expect us to be performing much, if any, external DNS traffic. (Some of our monitoring & metrics tools would require it, though, to reach their upstream providers.)
- Others:

**Comment by TomGeske** [comment]
+@palma21 can you take a look here?

**Comment by TomGeske** [comment]
I found this error message also upstream in coredns project.

**Comment by palma21** [comment]
Yes this normally would be a malformed response, the following timeout is interesting though. Could any of your traffic be blocked by any chance?
I think the best way here is to open a support ticket since it's likely related to this cluster only so we may also involve the folks from Azure DNS.

**Comment by curtdept** [comment]
@palma21  CoreDNS on k8s 1.19 seems to be 1.6.6 still? Shouldn't it be 1.7?

**Comment by palma21** [comment]
Unfortunately 1.7 is not backwards compatible so we're still evaluating it. (this should not be related to above OP thought, so happy to tackle it on a separate issue?)

**Comment by curtdept** [comment]
> Unfortunately 1.7 is not backwards compatible so we're still evaluating it. (this should not be related to above OP thought, so happy to tackle it on a separate issue?)

Only related in that there are several community reports 1.7 has some level of mitigation for these style of timeouts in addition of new concurrency limits which could help spread utilization more across instances and prevent latency spikes.

**Comment by palma21** [comment]
Thanks for reporting that, we'll take a look

**Comment by palma21** [comment]
FYI we confirmed the OP was caused by a DNS issue on that day. https://status.azure.com/en-us/status/history/

Creating a separate ticket for 1.7 and closing this one.
